crates/oxc_transformer/src/react/refresh.rs

```diff
@@ -1034,3 +966,38 @@ fn is_builtin_hook(hook_name: &str) -> bool {
         "useOptimistic"
     )
 }
+
+fn get_symbol_id_from_function_and_declarator(stmt: &Statement<'_>) -> Vec<SymbolId> {
+    let mut symbol_ids = vec![];
+    match stmt {
+        Statement::FunctionDeclaration(ref func) => {
+            symbol_ids.push(func.id.as_ref().unwrap().symbol_id.get().unwrap());
+        }
+        Statement::VariableDeclaration(ref decl) => {
+            symbol_ids.extend(decl.declarations.iter().filter_map(|decl| {
+                decl.id.get_binding_identifier().and_then(|id| id.symbol_id.get())
+            }));
+        }
+        Statement::ExportNamedDeclaration(ref export_decl) => {
+            if let Some(Declaration::FunctionDeclaration(func)) = &export_decl.declaration {
+                symbol_ids.push(func.id.as_ref().unwrap().symbol_id.get().unwrap());
+            } else if let Some(Declaration::VariableDeclaration(decl)) = &export_decl.declaration {
+                symbol_ids.extend(decl.declarations.iter().filter_map(|decl| {
+                    decl.id.get_binding_identifier().and_then(|id| id.symbol_id.get())
+                }));
+            }
+        }
+        Statement::ExportDefaultDeclaration(ref export_decl) => {
+            if let ExportDefaultDeclarationKind::FunctionDeclaration(func) =
+                &export_decl.declaration
+            {
+                if let Some(id) = func.id.as_ref() {
+                    symbol_ids.push(id.symbol_id.get().unwrap());
+                }
+            }
+        }
+        _ => {}
+    };
+
+    symbol_ids
+}
```

Comment:
According to https://github.com/oxc-project/oxc/issues/4767#issuecomment-2286162004. We determine where to insert new statements by `symbol_id`. If we have a Statement ID, it will be simpler to determine where we should insert

Reply:
Yes, very true.